### PR TITLE
rename hudson connector to jenkins connector #63

### PR DESCRIPTION
**Pull request by gnl42**
Replace references to Hudson with Jenkins. Existing Hudson repositories will be "converted" to Jenkins ones.

There are a couple of modules with references to Hudson that I don't know what to do about:

- 	org.eclipse.mylyn.jenkins.releng
-	org.eclipse.mylyn.mylyns.releng

I don't know where to get proper Jenkins png files so I left the Hudons ones alone.

org.eclipse.mylyn.discovery-directory/plugin.xml  references http://stats.mylyn.org/discovery/... but there doesn't seem to be anything there. Are the usage stats still collected?

### CODE REVIEW

**Comment by BeckerFrank** [comment]
@gnl42: You can ignore the two 
- org.eclipse.mylyn.jenkins.releng
- org.eclipse.mylyn.mylyns.releng

both have reverences to the old no longer maintained local test environment (vagrant setup). We now use the multipass way from org.eclipse.mylyn.releng/multipass

The domain mylyn.org is the actual location for the mylyn test instances used for pipeline builds, but I believe we no longer use the statistics service. Maybe @sddavis or @spingel can share some information.

**Comment by sddavis** [comment]
We at Tasktop/Planview are not collecting anything. As previously announced, we are not maintaining mylyn.org anymore and will be retiring it at some point. I would encourage active committers to make your own decisions about whether and how to collect usage stats.

**Comment by gnl42** [comment]
@sddavis Did the usage stats reveal anything interesting? Are/were they worth collecting?

**Comment by sddavis** [comment]
I don't remember what was collected or if I ever looked at it. It may have been before my time. I think it may just have been how many times each connector was installed from discovery.

File: org.eclipse.mylyn.releng/vagrant/Vagrantfile
Comment (BeckerFrank) [inline org.eclipse.mylyn.releng/vagrant/Vagrantfile:1]: We can remove the whole vagrant folder from org.eclipse.mylyn.releng

File: mylyn.builds/org.eclipse.mylyn.jenkins.releng/.project
Comment (BeckerFrank) [inline mylyn.builds/org.eclipse.mylyn.jenkins.releng/.project:1]: We can remove the whole project from mylyn.builds

**Review by BeckerFrank — COMMENTED** [review]
I think it is time to remove the vagrant setup.

**Comment by wimjongman** [comment]
Is this going to be moved to 3.27?

**Comment by ruspl-afed** [comment]
And this looks like "breaking change" since we rename features, bundles. API types. 
Do you think we can start with rename in UI and then do repackage gradually @gnl42 ?

**Comment by gnl42** [comment]
@ruspl-afed 
> And this looks like "breaking change" since we rename features, bundles. API types. Do you think we can start with rename in UI and then do repackage gradually @gnl42 ?

I haven't made much sense of what the issues are but I don't see a problem with just doing UI changes.
At the minimum, I can see breaking the move down to:

1. Visual changes. Text, icon
2. Plugin ID. Migrate from hudson config to jenkins config
3. Project/Package/Classes rename

**Comment by ruspl-afed** [comment]
> I haven't made much sense of what the issues are

It is simple. Imagine that some downstream product includes Mylyn Hudson for some time. Then, suddenly, after update to a service release everything stops working, because everything was repackaged from "hudson" to "jenkins".  [SemVer](https://semver.org) requires to do major segment increment for such cases, i.e. to have `3.26.0 -> 4.0.0` rather than `3.26.0` -> `3.26.0 -> 3.26.1`
It is more community friendly to announce such a change ahead of time and give people some time to migrate.

Regarding your suggestions:
> 1. Visual changes. Text, icon

Fine with me

> 2. Plugin ID. Migrate from hudson config to jenkins config

Feature ID is the key here. Plugin ID is also important, but Feature ID is how we go into other products.
We need to keep "hudson" Feature ID for some time (with deprecation mark on it), and at the same time introduce "jenkins"  Feature ID. Similar policy should be applied to Plugin ID

> 3. Project/Package/Classes rename

Similar with above. 
 * introduce "jenkins"
 * promote API carefully, to have less weight to carry
 * deprecate "hudson" and mark them for deletion

**Comment by mjmeijer** [comment]
Is this really worth spending precious time on?
Sounds like gut reacting to word 'hudson' while loving word 'jenkins'
+1 for UI improved labels/naming/description
-1 for package or feature/plugin ID changes

**Comment by akurtakov** [comment]
I would say yes as not many still remember the name "hudson" and feature/bundle names are visible in installation details of eclipse, also used by extenders (although these are supposed to know more of the history) and etc.

**Comment by wimjongman** [comment]
George, can you rebase so that we can get this in?

**Comment by gnl42** [comment]
Sure.

@wimjongman Side question, where can I put the attribution for the Jenkins image?

**Comment by wimjongman** [comment]
You mean org.eclipse.mylyn.jenkins.ui/icons/..  ?

**Comment by gnl42** [comment]
Yes. I'm grabbing the jenkins icon from https://www.jenkins.io/artwork/ which states:
` you must include an attribution to the Jenkins project, with a link (https://jenkins.io/), if you display these logos or derivates thereof`

**Comment by gnl42** [comment]
@BeckerFrank @ruspl-afed 
What does this mean:
`
R[INFO] Scanning for projects...
[ERROR] [ERROR] Project 'org.eclipse.mylyn:org.eclipse.mylyn.jenkins.feature:4.0.0-SNAPSHOT' is duplicated in the reactor @ 
[ERROR] Project 'org.eclipse.mylyn:org.eclipse.mylyn.jenkins.feature:4.0.0-SNAPSHOT' is duplicated in the reactor -> [1m[Help 1] `

**Comment by gnl42** [comment]
Never mind. Darn hudson-feature folder keeps re-appearing

**Comment by ruspl-afed** [comment]
Please let me know once you are ready for review here @gnl42

**Comment by gnl42** [comment]
It's been ready for  a while @ruspl-afed 

It keeps getting caught by all the other "global" changes

**Comment by ruspl-afed** [comment]
Yes, it was hard time to finish 3.26.1 and start 4.0.0
I plan to review this one soon.

Sorry for unrelated note: would you mind to open a discussion regarding future plans for JIRA integration support that you are doing [here](https://github.com/gnl42/JiraConnector)?

**Comment by ruspl-afed** [comment]
Looks good in general

One question. Where do we satisfy this requirement:
```
 you must include an attribution to the Jenkins project, with a link (https://jenkins.io/), if you display these logos or derivates thereof
```
?

**Comment by gnl42** [comment]
I added a comment to org.eclipse.mylyn.jenkins.ui/about.html. It was the best I could think of

File: mylyn.builds/org.eclipse.mylyn.jenkins.ui/about.html [hunk 1]
```diff
@@ -23,5 +23,8 @@ <h3>License</h3>
 indicated below, the terms and conditions of the EPL still apply to any source code in the Content
 and such source code may be obtained at <a href="/">http://www.eclipse.org</a>.</p>
 
+<p>
+Jenkins logo from <a href="https://jenkins.io">https://jenkins.io</a>
```
Comment (ruspl-afed) [inline mylyn.builds/org.eclipse.mylyn.jenkins.ui/about.html:27]: Ah, I missed it on the first viewing
 @BeckerFrank perhaps you can find and satisfy similar requirements for GitLab logo.

**Comment by wimjongman** [comment]
Wonderful!! Thanks, George and Alexander.

**Comment by merks** [comment]
I can now expect the next broken Oomph build. 😕   That's fine though.  Hudson died...

**Comment by wimjongman** [comment]
Yes, quite an embarrassing moment in the history of EF was the steal of Hudson.

**Comment by gnl42** [comment]
> Yes, it was hard time to finish 3.26.1 and start 4.0.0 I plan to review this one soon.
> 
> Sorry for unrelated note: would you mind to open a discussion regarding future plans for JIRA integration support that you are doing [here](https://github.com/gnl42/JiraConnector)?

Hi @ruspl-afed In terms of? 
At the moment the only plans I have is to replace the Bamboo code with a connector based on the Jenkins connector

**Pull request merged**